apps/hash-graph/lib/graph/src/store/postgres/knowledge/entity/read.rs

```diff
@@ -190,3 +192,123 @@ impl<C: AsClient> crud::Read<Entity> for PostgresStore<C> {
         Ok(stream)
     }
 }
+
+#[derive(Debug)]
+pub struct EntityEdgeTraversalData {
+    owned_by_ids: Vec<OwnedById>,
+    entity_uuids: Vec<EntityUuid>,
+    entity_revision_ids: Vec<Timestamp<VariableAxis>>,
+    intervals: Vec<RightBoundedTemporalInterval<VariableAxis>>,
+    resolve_depths: Vec<GraphResolveDepths>,
+    pinned_timestamp: Timestamp<PinnedAxis>,
+    variable_axis: TimeAxis,
+}
+
+impl EntityEdgeTraversalData {
+    pub const fn new(pinned_timestamp: Timestamp<PinnedAxis>, variable_axis: TimeAxis) -> Self {
+        Self {
+            owned_by_ids: Vec::new(),
+            entity_uuids: Vec::new(),
+            entity_revision_ids: Vec::new(),
+            intervals: Vec::new(),
+            resolve_depths: Vec::new(),
+            pinned_timestamp,
+            variable_axis,
+        }
+    }
+
+    pub fn push(
+        &mut self,
+        vertex_id: EntityVertexId,
+        interval: RightBoundedTemporalInterval<VariableAxis>,
+        resolve_depth: GraphResolveDepths,
+    ) {
+        self.owned_by_ids.push(vertex_id.base_id.owned_by_id);
+        self.entity_uuids.push(vertex_id.base_id.entity_uuid);
+        self.entity_revision_ids.push(vertex_id.revision_id);
+        self.intervals.push(interval);
+        self.resolve_depths.push(resolve_depth);
+    }
+}
+
+pub struct SharedEdgeTraversal {
+    pub left_endpoint: EntityVertexId,
+    pub right_endpoint: EntityTypeVertexId,
+    pub resolve_depths: GraphResolveDepths,
+    pub temporal_axes: QueryTemporalAxes,
+}
+
+impl<C: AsClient> PostgresStore<C> {
+    pub(crate) async fn read_shared_edges<'t>(
+        &self,
+        traversal_data: &'t EntityEdgeTraversalData,
+    ) -> Result<impl Iterator<Item = SharedEdgeTraversal> + 't, QueryError> {
+        let (pinned_axis, variable_axis) = match traversal_data.variable_axis {
+            TimeAxis::DecisionTime => ("transaction_time", "decision_time"),
+            TimeAxis::TransactionTime => ("decision_time", "transaction_time"),
+        };
+
+        Ok(self
+            .client
+            .as_client()
+            .query(
+                &format!(
+                    r#"
+                        SELECT
+                             filter.idx,
+                             ontology_ids.base_url,
+                             ontology_ids.version,
+                             filter.interval * source.{variable_axis}
```

Comment:
What is the `*` operation here?

Reply:
That's [the intersection](https://www.postgresql.org/docs/15/functions-range.html) of two ranges